client/src/main/java/io/pravega/client/stream/RetentionPolicy.java

```diff
@@ -57,6 +62,43 @@ public static RetentionPolicy byTime(Duration duration) {
      * @return Retention policy object.
      */
     public static RetentionPolicy bySizeBytes(long size) {
-        return new RetentionPolicy(RetentionType.SIZE, size);
+        return RetentionPolicy.builder().retentionType(RetentionType.SIZE).retentionParam(size).build();
+    }
+    
+    /**
+     * Create a retention policy to configure a stream to truncate a stream
+     * according to positions of subscribed reader groups.
+     *
+     * Provide null values for min and max if either limit is not desired.  
+     * @param type Type of consumption limit which is one of time or size. 
+     * @param minLimit min limit
+     * @param maxLimit max limit
+     * @return Retention policy object.
+     */
+    public static RetentionPolicy byConsumption(ConsumptionLimits.Type type, Long minLimit, Long maxLimit) {
+        return RetentionPolicy.builder().retentionType(RetentionType.CONSUMPTION)
+                              .consumptionLimits(ConsumptionLimits.builder().type(type)
+                                                                  .minValue(minLimit)
+                                                                  .maxValue(maxLimit)
+                                                                  .build()).build();
+    }
+    
+    @Data
+    @Builder
+    public static class ConsumptionLimits {
```

Comment:
Here we assume the user always wants to specify both values - min and max for a Stream.
Some users may just want to set a min value but not the max or vice versa.
It may be good to assign some placeholder value for a limit users don't want to exercise (perhaps -1 ??)